src/relay/op/tensor/transform.cc

```diff
@@ -3131,36 +3133,62 @@ bool MatrixSetDiagRel(const Array<Type>& types, int num_inputs, const Attrs& att
   const auto* diagonal = types[1].as<TensorTypeNode>();
   CHECK(diagonal);
 
+  const auto param = attrs.as<MatrixSetDiagAttrs>();
+  CHECK_GE(param->k2, param->k1);
+
```

Comment:
Could you also add a check to make sure the k values are in a sane range for the input tensor size?